llm/tests/test_cli.py

```
@@ -0,0 +1,240 @@
+
+def test_load_serving_config_rejects_missing_model_dir(tmp_path):
+    config_path = _write_config(
+        tmp_path,
+        {
+            "model": {},
+            "runtime": {"backend": "cpu"},
+            "generation": {},
+        },
+    )
+
+    with pytest.raises(ValueError, match="model.model_dir"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)